### PR TITLE
Polyfill for no IndexedDB support (fixes #165). r=gmarty

### LINKED ISSUE #165: Polyfill for no IndexedDB support
If IndexedDB is not supported by the browser, the app won't work at all. This impacts at least the following situations:
* Firefox in private mode
* iOS in web app (full screen app) mode (and webview probably), see #154

For such environments, we should use a in-memory storage instead of IndexedDB.

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

**Comment by azasypkin** [comment]
r? @gmarty 

How does it look to you? There are no unit tests here as I'm not sure if anyone needs this app in the future anyway :) App is deployed to my gh-pages.

File: app/js/lib/foxbox/common/in-memory-storage.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+const p = Object.freeze({ db: Symbol('db') });
```
Comment (gmarty) [inline app/js/lib/foxbox/common/in-memory-storage.js]: Can you start with `'use strict';` for the sake of consistency?

File: app/js/lib/foxbox/db.js [hunk 2]
```diff
@@ -25,31 +18,44 @@ const DB_VERSION = 1;
 
 const DB_SERVICE_STORE = 'services';
 
+class FoxboxIndexedDbStorage extends IndexedDbStorage {
```
Comment (gmarty) [inline app/js/lib/foxbox/db.js]: I'd rather call it something like `FoxBoxDbStorage` as it isn't always backed by IndexedDB.
What do you think?

File: app/js/lib/foxbox/common/indexed-db-storage.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
+const p = Object.freeze({
```
Comment (gmarty) [inline app/js/lib/foxbox/common/indexed-db-storage.js]: nit: same here.

File: app/js/lib/foxbox/db.js (hunk 2 above)
Comment (gmarty) [inline app/js/lib/foxbox/db.js]: Got it. Please ignore my comment.

**Comment by gmarty** [comment]
It looks good to me. r+
Excellent job!

File: app/js/lib/foxbox/common/in-memory-storage.js (hunk 1 above)
Comment (azasypkin) [inline app/js/lib/foxbox/common/in-memory-storage.js]: Ah, yeah, sure

**Comment by azasypkin** [comment]
Thanks for review! Nits fixed, checked the case when IndexedDB is null (just to be 100% sure) and waiting for travis.

**Pull request merged**